java/springboot_hikaricp_using_dsql_connector/src/main/java/org/example/Example.java

```diff
@@ -0,0 +1,241 @@
+        if (conn == null) {
+            throw new SQLException("Connection cannot be null");
+        }
+        
+        //cleanup
+        //dropSampleTable(conn);
+
+        // Create a new table named owner (Aurora DSQL compatible)
+        Statement create = conn.createStatement();
+        if (create != null) {
+            create.executeUpdate("""
+                    CREATE TABLE IF NOT EXISTS owner(
+                    id uuid NOT NULL DEFAULT gen_random_uuid(),
+                    name varchar(30) NOT NULL,
+                    city varchar(80) NOT NULL,
+                    telephone varchar(20) DEFAULT NULL)""");
+            create.close();
+        }
+
+        // Insert some data with a unique identifier
+        String uniqueName = "John Doe " + System.currentTimeMillis() + "_" + connectionNumber;
+        Statement insert = conn.createStatement();
+        if (insert != null) {
+            insert.executeUpdate(
+                    "INSERT INTO owner ( name, city, telephone) VALUES ( '" + uniqueName + "', 'Anytown', '555-555-1991')");
+            insert.close();
+        }
+
+        // Read back the data and verify
+        String selectSQL = "SELECT * FROM owner WHERE name = '" + uniqueName + "'";
+        Statement read = conn.createStatement();
+        if (read != null) {
+            ResultSet rs = read.executeQuery(selectSQL);
+            if (rs != null) {
+                while (rs.next()) {
+                    String id = rs.getString("id");
+                    String name = rs.getString("name");
+                    String city = rs.getString("city");
+                    String telephone = rs.getString("telephone");
+                    
+                    assert id != null;
+                    assert name != null && name.equals(uniqueName);
+                    assert city != null && city.equals("Anytown");
+                    assert telephone != null && telephone.equals("555-555-1991");
+                    System.out.println("Data verified: " + name + " from " + city + " (ID: " + id + ")");
+                }
+            }
+            read.close();
+        }
+    }
+
+    public static void main(String[] args) throws SQLException {
+        System.out.println("Starting Aurora DSQL Connection Manager with HikariCP and Aurora DSQL JDBC Connector");
+        System.out.println();
+
+        // Environment variables for configuration
+        String clusterEndpoint = System.getenv("CLUSTER_ENDPOINT");
+        String clusterUser = System.getenv("CLUSTER_USER");
+
+        // Validation
+        assert clusterUser != null : "CLUSTER_USER environment variable is not set";
+        assert clusterEndpoint != null && !clusterEndpoint.isEmpty() : "CLUSTER_ENDPOINT environment variable is not set";
+
+        // Build the JDBC URL
+        String url = "jdbc:aws-dsql:postgresql://" + clusterEndpoint + "/postgres?user=" + clusterUser;
```

Comment:
Why do we need a URL here given the `application.yml` specifies both the endpoint and the username already?